### PR TITLE
Alertmanager: Add HPA config to jsonnet lib

### PR DESCRIPTION
**Pull request by JohnnyQQQQ**
#### What this PR does

This PR adds the auto scaler config to the mixin for the Alertmanager. The config has been battle-tested during the last few months at Grafana Labs.

#### Checklist

- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
 ### Jsonnet
 
 * [ENHANCEMENT] Add per-container map for environment variables. #5181
+* [FEATURE] Alertmanager: Add horizontal pod autoscaler config, that can be enabled using `autoscaling_alertmanager_enabled: true`.
```
Comment (stevesg) [inline CHANGELOG.md:21]: [nit] Features come before enhancements, add have PR number.

```suggestion
* [FEATURE] Alertmanager: Add horizontal pod autoscaler configuration, that can be enabled using `autoscaling_alertmanager_enabled: true`. #5194
* [ENHANCEMENT] Add per-container map for environment variables. #5181
```

File: operations/mimir/autoscaling.libsonnet [hunk 2]
```diff
@@ -419,4 +424,54 @@
...
+      min_replicas=$._config.autoscaling_alertmanager_min_replicas,
+      max_replicas=$._config.autoscaling_alertmanager_max_replicas,
+      memory_target_utilization=$._config.autoscaling_alertmanager_memory_target_utilization,
+    ) + {
+      spec+: {
+        scaleTargetRef+: {
+          kind: 'StatefulSet',
+        },
+      },
+    },
```
Comment (stevesg) [inline operations/mimir/autoscaling.libsonnet:476]: [super-nit] These are defined the other way round for other components - do the same for consistency.

```suggestion
  alertmanager_scaled_object: if !$._config.autoscaling_alertmanager_enabled then null else
    $.newAlertmanagerScaledObject(
      name='alertmanager',
      alertmanager_cpu_requests=$.alertmanager_container.resources.requests.cpu,
      alertmanager_memory_requests=$.alertmanager_container.resources.requests.memory,
      min_replicas=$._config.autoscaling_alertmanager_min_replicas,
      max_replicas=$._config.autoscaling_alertmanager_max_replicas,
      memory_target_utilization=$._config.autoscaling_alertmanager_memory_target_utilization,
    ) + {
      spec+: {
        scaleTargetRef+: {
          kind: 'StatefulSet',
        },
      },
    },

  alertmanager_statefulset: overrideSuperIfExists(
    'alertmanager_statefulset',
    if !$._config.autoscaling_alertmanager_enabled then {} else removeReplicasFromSpec
  ),
```

**Review by stevesg — APPROVED** [review]
LGTM, just nits.

FWIW To other reviewers - this is the configuration we have been running at Grafana Labs for quite a period of time.

**Review by 56quarters — APPROVED** [review]
LGTM modulo @stevesg's comments

File: CHANGELOG.md (hunk 1 above)
Comment (JohnnyQQQQ) [reply CHANGELOG.md:21]: Changed.

File: operations/mimir/autoscaling.libsonnet (hunk 2 above)
Comment (JohnnyQQQQ) [reply operations/mimir/autoscaling.libsonnet]: Changed.

**Comment by JohnnyQQQQ** [comment]
Thanks for the review. I have addressed the feedback 👍

**Pull request merged**